### PR TITLE
Initial backend setup (please ignore the login/signup part, as I will make a separate PR for those)

### PR DESCRIPTION
**Pull request by CristobalMe**

### CODE REVIEW

File: backend/prisma/migrations/20240628180651_data_model_v1/migration.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
+-- CreateTable
+CREATE TABLE "event" (
```
Comment (ygongdev) [inline backend/prisma/migrations/20240628180651_data_model_v1/migration.sql:2]: Just wondering if the file name is intented, `migrations/20240628180651_data_model_v1/migration.sql` sounds kind of weird

File: frontend/src/App.jsx [hunk 2]
```diff
@@ -1,22 +1,58 @@
-import './App.css'
 import HomePage from "./components/homePage/HomePage.jsx"
```
Comment (ygongdev) [inline frontend/src/App.jsx:1]: If possible, let's keep frontend and backend changes as separate commits or PRs. So when you're developing, you are mostly likely making both backend and frontend changes. So when creating the commit, you can selectively add backend changes into the first commit and frontend changes into the second commit since it is most likely that your frontend changes is dependent on your backend changes.

File: frontend/src/App.jsx [hunk 3]
```diff
@@ -1,22 +1,58 @@
...
+
+
 function App() {
 
+  const [user, setUser] = useState(() => {
+    // Retrieve the user data from storage or set it to null if not found
+    const storedUser = localStorage.getItem('user');
+    // return storedUser ? JSON.parse(storedUser) : null;
+    return null
+  });
```
Comment (ygongdev) [inline frontend/src/App.jsx:24]: Does this actually work? You're initializing `user` to be a function that returns `storedUser`. I'm assuming you want it to be the actual user, not a function.
So initially, it should be `const [user, setUser] = useState(null);`

If you want to try to see if you already stored the user in local storage and fetching it on initialization, you can do
```javascript
useEffect(() => {
  const storedUser = localStorage.getItem('user');
  setUser(storedUser);
}, []);
```

Comment (ygongdev) [reply frontend/src/App.jsx:24]: Or even better

```
const [user, setUser] = useState(localStorage.getItem('user') ?? null);
```

File: frontend/src/components/LoginForm/LoginForm.jsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  const { updateUser } = useContext(UserContext);
+
+  const navigate = useNavigate();
+
+  const handleLogin = async (e) => {
+    e.preventDefault();
+
+    try {
+      // Make the login API request
+      const response = await fetch(`http://localhost:3000/users/login`, {
```
Comment (ygongdev) [inline frontend/src/components/LoginForm/LoginForm.jsx:18]: Probably want to make sure this will support production use case too instead of just `localhost`.

File: frontend/src/components/LoginForm/LoginForm.jsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+          <label htmlFor="password">Password:</label>
+          <input
+            type="password"
+            id="password"
+            value={password}
+            onChange={(e) => setPassword(e.target.value)}
+            required
+          />
+        </div>
+        <button type="submit">Login</button>
```
Comment (ygongdev) [inline frontend/src/components/LoginForm/LoginForm.jsx:70]: What would happen if the user left `username` and `password` blank? Can we make this experience better, e.g disable if empty?

File: frontend/src/components/SignupForm/SignupForm.jsx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+  const { updateUser } = useContext(UserContext);
+  const navigate = useNavigate();
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    try {
+      // Make the signup API request
+      const response = await fetch(`http://localhost:3000/users`, {
```
Comment (ygongdev) [inline frontend/src/components/SignupForm/SignupForm.jsx:19]: same here about `localhost` in production

File: frontend/src/components/SignupForm/SignupForm.jsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
+import React, { useState, useContext } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import './SignupForm.css'
+import { UserContext } from '../../UserContext.js';
+
+const SignupForm = () => {
+  const [username, setUsername] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
```
Comment (ygongdev) [inline frontend/src/components/SignupForm/SignupForm.jsx:9]: Potential Advanced Feature Opportunity:
1. Some kind of email format validation, so that entries like `foo@bar` can't be used as email
2. Enforce pattern matching for password, e.g must have a symbol
3. Ensure all usernames are unique, e.g username has been taken, please choose another.

Comment (ygongdev) [inline frontend/src/components/SignupForm/SignupForm.jsx:9]: Another potential advanced feature opportunity:
1. Improve password security by salt and hashing (if you haven't looked into it).

File: backend/prisma/migrations/20240628180651_data_model_v1/migration.sql (hunk 1 above)
Comment (CristobalMe) [reply backend/prisma/migrations/20240628180651_data_model_v1/migration.sql:2]: Yes, its the name that prisma gives to the migration. I will try to change it to see what happens

File: frontend/src/App.jsx (hunk 3 above)
Comment (CristobalMe) [reply frontend/src/App.jsx:24]: Thank you. I was having problems with that

File: frontend/src/components/LoginForm/LoginForm.jsx (hunk 5 above)
Comment (CristobalMe) [reply frontend/src/components/LoginForm/LoginForm.jsx:70]: Screenshot 2024-07-01 at 9 32 08 AM

File: frontend/src/components/SignupForm/SignupForm.jsx (hunk 7 above)
Comment (CristobalMe) [reply frontend/src/components/SignupForm/SignupForm.jsx:9]: To do: 1. & 2.

File: frontend/src/components/LoginForm/LoginForm.jsx (hunk 5 above)
Comment (CristobalMe) [reply frontend/src/components/LoginForm/LoginForm.jsx:70]: Done

File: frontend/src/components/LoginForm/LoginForm.jsx (hunk 4 above)
Comment (CristobalMe) [reply frontend/src/components/LoginForm/LoginForm.jsx:18]: To do

File: frontend/src/App.jsx (hunk 3 above)
Comment (CristobalMe) [reply frontend/src/App.jsx:24]: To do

File: frontend/src/App.jsx (hunk 2 above)
Comment (CristobalMe) [reply frontend/src/App.jsx:1]: Done

File: frontend/src/App.jsx (hunk 3 above)
Comment (CristobalMe) [reply frontend/src/App.jsx:24]: Done

File: frontend/src/components/LoginForm/LoginForm.jsx (hunk 4 above)
Comment (CristobalMe) [reply frontend/src/components/LoginForm/LoginForm.jsx:18]: Done

File: frontend/src/components/SignupForm/SignupForm.jsx (hunk 6 above)
Comment (CristobalMe) [reply frontend/src/components/SignupForm/SignupForm.jsx:19]: Done
